src/full/Agda/TypeChecking/Pretty.hs

```diff
@@ -4,6 +4,14 @@
 {-# LANGUAGE TupleSections        #-}
 {-# LANGUAGE UndecidableInstances #-}
 
+-- To define <>, we will probably need to add:
+--import Prelude hiding ((<>))
+-- but using that now gives warnings and doesn't silence -Wsemigroup
+#if __GLASGOW_HASKELL__ >= 800
+{-# OPTIONS_GHC -Wno-semigroup    #-}
+#endif
```

Comment:
Thanks for the link! That means, hopefully, that before 8.2 this might be solved.
Or do you suggest an alternative solution here (though as a separate commit, I guess)?

In that direction, I now wonder whether this file could overload `(<>)` by defining a `Semigroup` instance, but it would have the wrong fixity.

However, this `(<>)` *seems* to also have an imperfect fixity, it seems to take the default `infixl 9` (had to check https://www.haskell.org/onlinereport/decls.html) instead of `pretty`'s `infixl 6`.